### PR TITLE
add support for x86_64 and libvirt using virt-install

### PR DESCRIPTION
**Pull request by alknopfler**
add support for x86_64 and libvirt using virt-install
Features:
- Autodiscovery host architecture to use virt-install or utm based on the architecture
- keep the combustion out of the switch part to re-use it for x86_64 hosts

### CODE REVIEW

**Review by e-minguez — CHANGES_REQUESTED** [review]
Added a couple of suggestions

File: slemicro/create_vm.sh [hunk 1]
```diff
@@ -82,55 +88,71 @@ mkisofs -quiet -full-iso9660-filenames -o ${VMFOLDER}/ignition-and-combustion-${
...
+      if [[ ${count} -ge ${timeout} ]]; then
+        break
+      fi
+      sleep 1
+      echo -n "."
+  done
+  VMIP=$(virsh domifaddr ${VMNAME} | grep ipv | awk '{print $4}'| cut -d"/" -f 1 )
+else # Darwin
+  # Create and launch the VM using UTM
+  OUTPUT=$(osascript <<END
```
[the remark this answers is not in the document]
Comment (alknopfler) [reply slemicro/create_vm.sh:109]: done

File: slemicro/create_vm.sh [hunk 2]
```diff
@@ -82,55 +88,71 @@ mkisofs -quiet -full-iso9660-filenames -o ${VMFOLDER}/ignition-and-combustion-${
...
 
-OUTPUT=$(
-osascript <<END
-tell application "UTM"
-	set vm to virtual machine named "${VMNAME}"
-	set config to configuration of vm
-	get address of item 1 of network interfaces of config
-end tell
+  OUTPUT=$(
+  osascript <<END
```
[the remark this answers is not in the document]
Comment (alknopfler) [reply slemicro/create_vm.sh:129]: done

File: slemicro/create_vm.sh [hunk 3]
```diff
@@ -10,12 +10,21 @@ die(){
 # Get the env file
 source ${BASEDIR}/.env
 
+if [ $# -eq 1 ]; then
+    VMNAME=$1
```
Comment (e-minguez) [inline slemicro/create_vm.sh:14]: Please modify the README to show how to use this argument :)

File: slemicro/combustion/script [hunk 4]
```diff
@@ -385,7 +385,7 @@ if [ "${ELEMENTAL}" = true ]; then
 	    name: elemental
 	    noCache: false
 	    version: 1.1.0
-	FOO
+FOO
```
Comment (e-minguez) [inline slemicro/combustion/script:388]: This should be keep as it is. PTAL

**Review by e-minguez — CHANGES_REQUESTED** [review]
Some changes required :)

File: slemicro/create_vm.sh [hunk 5]
```diff
@@ -82,55 +91,71 @@ mkisofs -quiet -full-iso9660-filenames -o ${VMFOLDER}/ignition-and-combustion-${
...
-	set vm to make new virtual machine with properties {backend:qemu, configuration:{cpu cores:${CPUS}, memory: ${MEMORY}, name:"${VMNAME}", architecture:"aarch64", drives:{{removable:true, source:iso}, {removable:false, source:rawfile}}}}
-	start vm
-	repeat
-		if status of vm is started then exit repeat
-	end repeat
-	get address of first serial port of vm
-end tell
+# if x86_64, convert the image to qcow2 to be used by kcli
+if [ $(uname -o) == "GNU/Linux" ]; then
+  qemu-img convert -O qcow2 ${VMFOLDER}/${VMNAME}.raw ${VMFOLDER}/${VMNAME}.qcow2
```
Comment (e-minguez) [inline slemicro/create_vm.sh:96]: You should also do this https://github.com/suse-edge/misc/blob/main/slemicro/create_vm.sh#L48 to grow the disk as you want.

Comment (alknopfler) [reply slemicro/create_vm.sh:96]: the part you said in line 48 is common to both, this part is only to convert it.

File: slemicro/create_vm.sh [hunk 6]
```diff
@@ -82,55 +91,71 @@ mkisofs -quiet -full-iso9660-filenames -o ${VMFOLDER}/ignition-and-combustion-${
...
-	set rawfile to POSIX file "${VMFOLDER}/${VMNAME}.raw"
-	--- create a new QEMU VM 
-	set vm to make new virtual machine with properties {backend:qemu, configuration:{cpu cores:${CPUS}, memory: ${MEMORY}, name:"${VMNAME}", architecture:"aarch64", drives:{{removable:true, source:iso}, {removable:false, source:rawfile}}}}
-	start vm
-	repeat
-		if status of vm is started then exit repeat
-	end repeat
-	get address of first serial port of vm
-end tell
+# if x86_64, convert the image to qcow2 to be used by kcli
```
Comment (e-minguez) [inline slemicro/create_vm.sh:94]:
```suggestion
# if x86_64, convert the image to qcow2
```

File: slemicro/create_vm.sh [hunk 7]
```diff
@@ -82,55 +91,71 @@ mkisofs -quiet -full-iso9660-filenames -o ${VMFOLDER}/ignition-and-combustion-${
...
+  while [ $(virsh domifaddr ${VMNAME} | grep ipv | awk '{print $4}'| cut -d"/" -f 1 | wc -l) -ne 1 ]; do
+      count=$((count + 1))
+      if [[ ${count} -ge ${timeout} ]]; then
+        break
+      fi
+      sleep 1
+      echo -n "."
+  done
+  VMIP=$(virsh domifaddr ${VMNAME} | grep ipv | awk '{print $4}'| cut -d"/" -f 1 )
+else # Darwin
```
Comment (e-minguez) [inline slemicro/create_vm.sh:110]:
```suggestion
elif [ $(uname -o) == "Darwin" ]; then
```
For consistency with the other `if`

File: slemicro/delete_vm.sh [hunk 8]
```diff
@@ -9,22 +9,29 @@ die(){
...
+	  if status of vm is started then stop vm
+	  repeat
+	    if status of vm is stopped then exit repeat
+    end repeat
+	  delete virtual machine named "${VMNAME}"
+  end tell
 END
-)
+  )
+else
```
Comment (e-minguez) [inline slemicro/delete_vm.sh:31]:
```suggestion
elif [ $(uname -o) == "GNU/Linux" ]; then
```
For consistency with the other `if`

**Review by e-minguez — CHANGES_REQUESTED** [review]
Minor stuff :)
Also, it would be nice to keep the `ifs` the same. The first one says if os = linux, then always keep the same comparision.

File: slemicro/delete_vm.sh [hunk 9]
```diff
@@ -9,22 +9,32 @@ die(){
...
-tell application "UTM"
-	set vm to virtual machine named "${VMNAME}"
-	if status of vm is started then stop vm
-	repeat
-	  if status of vm is stopped then exit repeat
-  end repeat
-	delete virtual machine named "${VMNAME}"
-end tell
+  OUTPUT=$(
+  osascript <<END
```
Comment (e-minguez) [inline slemicro/delete_vm.sh]:
```suggestion
  osascript <<-END
```

File: slemicro/create_vm.sh [hunk 10]
```diff
@@ -82,55 +91,73 @@ mkisofs -quiet -full-iso9660-filenames -o ${VMFOLDER}/ignition-and-combustion-${
...
+    set rawfile to POSIX file "${VMFOLDER}/${VMNAME}.raw"
+    --- create a new QEMU VM
+    set vm to make new virtual machine with properties {backend:qemu, configuration:{cpu cores:${CPUS}, memory: ${MEMORY}, name:"${VMNAME}", architecture:"aarch64", drives:{{removable:true, source:iso}, {removable:false, source:rawfile}}}}
+    start vm
+    repeat
+      if status of vm is started then exit repeat
+    end repeat
+    get address of first serial port of vm
+  end tell
 END
```
Comment (e-minguez) [inline slemicro/create_vm.sh:126]: This needs to be tabbed as well

File: slemicro/create_vm.sh [hunk 11]
```diff
@@ -82,55 +91,73 @@ mkisofs -quiet -full-iso9660-filenames -o ${VMFOLDER}/ignition-and-combustion-${
...
-	get address of item 1 of network interfaces of config
-end tell
+  OUTPUT=$(
+  osascript <<-END
+  tell application "UTM"
+    set vm to virtual machine named "${VMNAME}"
+    set config to configuration of vm
+    get address of item 1 of network interfaces of config
+  end tell
 END
```
Comment (e-minguez) [inline slemicro/create_vm.sh:138]: This needs to be tabbed as well

File: slemicro/delete_vm.sh [hunk 12]
```diff
@@ -9,22 +9,32 @@ die(){
...
+  osascript <<-END
+  tell application "UTM"
+	  set vm to virtual machine named "${VMNAME}"
+	  if status of vm is started then stop vm
+	  repeat
+	    if status of vm is stopped then exit repeat
+    end repeat
+	  delete virtual machine named "${VMNAME}"
+  end tell
 END
```
Comment (e-minguez) [inline slemicro/delete_vm.sh:29]: This needs to be tabbed as well

File: slemicro/create_vm.sh [hunk 13]
```diff
@@ -82,55 +91,73 @@ mkisofs -quiet -full-iso9660-filenames -o ${VMFOLDER}/ignition-and-combustion-${
...
-	--- create a new QEMU VM 
-	set vm to make new virtual machine with properties {backend:qemu, configuration:{cpu cores:${CPUS}, memory: ${MEMORY}, name:"${VMNAME}", architecture:"aarch64", drives:{{removable:true, source:iso}, {removable:false, source:rawfile}}}}
-	start vm
-	repeat
-		if status of vm is started then exit repeat
-	end repeat
-	get address of first serial port of vm
-end tell
+# if x86_64, convert the image to qcow2
+if [ $(uname -o) == "GNU/Linux" ]; then
```
Comment (e-minguez) [inline slemicro/create_vm.sh:95]: This should be Darwin first, elif GNU/Linux (to be consistent with the first `if`)

**Review by e-minguez — CHANGES_REQUESTED** [review]
Sorry! 😅 More changes requested.

**Review by e-minguez — CHANGES_REQUESTED** [review]
Sorry! 😅 More changes requested.

**Comment by alknopfler** [comment]
all changed...appears outdated!!!

**Pull request merged**